packages/app-shell-shared/src/types/Config.ts

```diff
@@ -20,7 +40,7 @@ export type HvAppShellMenuConfig = {
   icon?: HvAppShellIcon;
   target?: string;
   submenus?: HvAppShellMenuConfig[];
-};
+} & HvAppShellConditionalConfig;
```

Comment:
let's convert these to `interface` with `extends` instead, which is significantly faster (than `&`) makes it user extendable

overall, `interface` is better (when it's an option), but `type` as more features